.github/workflows/greetings.yml

```diff
@@ -1,85 +1,64 @@
-                ;;
-            esac
-          fi
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-      
-      - name: Comment on Issue
-        if: ${{ github.event_name == 'issues' }}
-        run: |
-          actor_type="${{ steps.determine_actor_type.outputs.actor_type }}"
-          issue_message=""
-          if [[ "${{ github.actor }}" != "dependabot[bot]" ]]; then
-            case $actor_type in
-              "contributor")
-                issue_message="Hello ${{ github.actor }}, thanks for opening an issue! Your contribution is valuable to us. The maintainers will review this issue and provide feedback as soon as possible."
-                ;;
-              *)
-                issue_message="Hello ${{ github.actor }}! Thank you for your interaction with this repository. We have noted your issue."
-                ;;
-            esac
-            echo "Issue Comment: $issue_message"
-          fi
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-      
-      - name: Comment on Pull Request
-        if: ${{ github.event_name == 'pull_request_target' }}
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: 14
+
+      - name: Install dependencies
+        run: npm install
+
+      - name: Greet Contributor
+        id: greet
         run: |
-          actor_type="${{ steps.determine_actor_type.outputs.actor_type }}"
-          pr_message=""
-          if [[ "${{ github.actor }}" != "dependabot[bot]" ]]; then
-            case $actor_type in
-              "contributor")
-                pr_message="Hello ${{ github.actor }}, thanks for raising a pull request! Your contribution is valuable to us. The maintainers will review this pull request and provide feedback as soon as possible. Keep up the great work!"
-                ;;
-              *)
-                pr_message="Hello ${{ github.actor }}! Thank you for your interaction with this repository. We have noted your pull request."
-                ;;
-            esac
-            echo "Pull Request Comment: $pr_message"
-          fi
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          echo "::set-output name=greeting::$(node -e '
+          // Greetings mapping
+          const greetings = {
+            collaborator: "Hello, collaborator! Thank you for your contribution. We appreciate your help!",
+            owner: "Welcome, owner! We're glad to have you on board. Your expertise is valuable to us!",
+            firstTimeContributor: "Hello, first-time contributor! Thank you for opening your first issue or pull request. We're excited to collaborate with you!",
+            returningContributor: "Welcome back, returning contributor! Your continued contributions are greatly appreciated!",
```

Comment:
I think there is scope for improvement in the message. See i Don't think they are ok. and also greeting way would be different....We should write hello firstTimeContributor... instead the name but we have to greet as they comes for the first. I hope you get the point